### DISCUSSION TITLE
Get the original page route in components.

### DISCUSSION THREAD

**Discussion by hota911** [vercel/next.js#49685] [Help] [answered]
### Summary

Hi,

I'm writing an application in Next v12.
I'd like to get the original page route in components.
For example, when rendering `pages/blog/[slug].ts` I would like to get `"blog/[slug]"` not the actual path `blog/a`.
Is there any way to do so?

**Comment by lucaspiresfernandes** [comment, marked as answer]
pages/foo/[bar]
``` ts
import { useRouter } from "next/router"

export default function Page() {
    const router = useRouter();
    console.log(router.pathname);
    return <></>;
}
```

Logs `/foo/[bar]`.

**Reply by hota911** [reply]
Thank you!
